### ISSUE TITLE
client.files.retrieve_content only returns strings, and not bytes/binary data

### ISSUE THREAD

**Issue by eware-godaddy** [openai/openai-python#699]
When I try to download/retrieve a binary file (eg. PNG image) created by an assistant, it get automatically [cast to a string](https://github.com/openai/openai-python/blob/e0aafc6c1a45334ac889fe3e54957d309c3af93f/src/openai/resources/files.py#L229), so it can't be correctly parsed/displayed.

Eg:

``` py
ret_file = client.files.retrieve_content('file-XXX')
ret_file[:10]
# '�PNG\r\n\x1a\n\x00\x00'
```

There doesn't seem to be a clean way in the API to retrieve a file as raw bytes from what I can see.

This is important for code interpreter scenarios where the agent returns binary files that need to be rendered, like Images.

For others having this issue, you can just request the files directly using `requests` like:

``` py
import requests
from io import BytesIO

file_id = 'file-XXXX'
headers = {
    'Authorization': f"Bearer {os.environ['OPENAI_API_KEY']}"
}
response = requests.get(f'https://api.openai.com/v1/files/{file_id}/content', headers=headers)
Image.open(BytesIO(response.content))
```

**Comment by tnm** [comment]
Hit this as well, and also using requests directly. Related, the general OpenAI API documentation for this should probably cover the format/expectations of what's coming back

**Comment by RobertCraigie** [comment]
Note that you do not have to drop down to using raw `requests` here, the SDK exposes methods for accessing the raw response directly using `.with_raw_response` properties. e.g.

```py
response = client.files.with_raw_response.retrieve_content('file-XXX')
response  # APIResponse
response.content  # raw bytes
```

**Comment by RobertCraigie** [comment]
This has been fixed in the `v1.2.1` release, you can now use `client.files.content()` instead and `client.files.retrieve_content()` has been deprecated.
